lib/Partitioner/Partitioner.cpp

```diff
@@ -258,17 +258,69 @@ void Partitioner::initOpComputeTime(Function *F) {
       /// compute total bytes read
       uint64_t sizeInput = numLookups * sizePerLookup;
 
-      /// does the table fit in SRAM or DRAM
-      if (tableSize > sramCapacity) {
-        sizeDram += sizeInput;
-      } else {
-        sizeSram += sizeInput;
-      }
-
+      /// tables are usually large and fit in DRAM
+      sizeDram += sizeInput;
       /// we also read the indices, weights and lengths arrays
       sizeSram += SLWSN->getIndices().getType()->getSizeInBytes();
       sizeSram += SLWSN->getWeights().getType()->getSizeInBytes();
       sizeSram += SLWSN->getLengths().getType()->getSizeInBytes();
+    } else if (node.getKind() == Kinded::Kind::SparseLengthsSumNodeKind) {
```

Comment:
there is a lot of duplication in FusedRowwiseQuantizedSparseLengthsWeightedSumNodeKind and other weighted variations. Similar in non weighted. Can this be extracted into a template method to reuse the code.

Reply:
Discussed offline and agreed to defer to the restructuring pass.